js/treeViewPager.js

```diff
@@ -0,0 +1,105 @@
+"use strict";
+
+(function ($, Qubit) {
+
+  Qubit.TreeviewPager = function(limit, treeEl, url)
+  {
+    Qubit.Pager.call(this, limit);
+    this.treeEl = treeEl;
+    this.url = url;
+    this.rootId = '#';
+  }
+
+  Qubit.TreeviewPager.prototype = new Qubit.Pager;
+
+  Qubit.TreeviewPager.prototype.getAndAppendNodes = function(cb)
+  {
+    // Assemble query and creation queue
+    var queryString = "?skip=" + this.getSkip() + "&nodeLimit=" + this.getLimit();
+    var pagedUrl = this.url + queryString;
+    var createQueue = [];
+    var self = this;
+
+    // Get and append additional nodes
+    $.ajax({
+      url: pagedUrl,
+      success: function(results) {
+
+        // Add nodes to creation queue
+        results.nodes.forEach(function(node) {
+          createQueue.push(node);
+        });
+
+        var next = function()
+        {
+          if (createQueue.length)
+          {
+            // Queue isn't empty: create node
+            var node = createQueue.shift();
+            self.treeEl.jstree(true).create_node(self.rootId, node, "last", next);
+          }
+          else
+          {
+            // Queue is empty so excute cleanup logic
+            cb();
+          };
+        };
+
+        next();
+      }
+    });
+  }
+
+  // Update count of remaining nodes, etc.
+  Qubit.TreeviewPager.prototype.updateMoreLink = function($moreButton, $resetButton)
+  {
+    var scrollOffset = 0;
+
+    if (this.getRemaining() > 0)
+    {
+      // Update count shown in paging button
+      $moreButton.show();
+      $moreButton.val($moreButton.data('label').replace('%1%', this.getRemaining()));
```

Comment:
Not a big deal but, it's possible to change the value before calling `show()`?

Reply:
Good thinking.